src/client/Microsoft.Identity.Client/ApiConfig/Executors/ManagedIdentityExecutor.cs

```diff
@@ -35,19 +38,44 @@ public ManagedIdentityExecutor(IServiceBundle serviceBundle, ManagedIdentityAppl
         {
             var requestContext = CreateRequestContextAndLogVersionInfo(commonParameters.CorrelationId, cancellationToken);
 
+            if (_managedIdentityApplication.KeyMaterialManager.CryptoKeyType != CryptoKeyType.None)
+            {
+                // Check resource format 
+                if (Uri.TryCreate(managedIdentityParameters.Resource, UriKind.Absolute, out Uri resourceUri) &&
+                    (resourceUri.Scheme == Uri.UriSchemeHttp || resourceUri.Scheme == Uri.UriSchemeHttps) &&
+                    !managedIdentityParameters.Resource.EndsWith("/.default", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Add "/.default" to the scopes
+                    commonParameters.Scopes = new SortedSet<string>
+                    {
+                        managedIdentityParameters.Resource + "/.default"
+                    };
+                }
+
+                requestContext.Logger.Info(LogMessages.CredentialScopeUpdated);
```

Comment:
Generally we preferer not have have constants when logging. Makes the code easier to read.